mteb/evaluation/evaluators/PairClassificationEvaluator.py

```diff
@@ -72,6 +72,34 @@ def __call__(
         scores["main_score"] = main_score
         return scores
 
+    @staticmethod
+    def _encode_unique_texts(
+        all_texts: list[str],
+        model: Encoder,
+        task_name: str | None,
+        prompt_type: PromptType | None,
+        **encode_kwargs: Any,
+    ):
+        index_map, all_unique_texts, all_texts_indexes = {}, [], []
+        for text in all_texts:
+            text_hash = hash(text)
+            if text_hash not in index_map:
+                index_map[text_hash] = len(all_unique_texts)
+                all_unique_texts.append(text)
+            all_texts_indexes.append(index_map[text_hash])
+        logger.warning(
+            f"A total on {len(all_texts) - len(all_unique_texts)}/{len(all_texts)} duplicate texts were found during encoding. Only encoding unique text and duplicating embeddings across."
+        )
+        all_unique_texts_embs = np.asarray(
+            model.encode(
+                all_unique_texts,
+                task_name=task_name,
+                prompt_type=prompt_type,
```

Comment:
I think prompt type can be removed from evaluator

Reply:
I'm not sure I follow. This is from the Encoder interface. The AbsTask also has a task prompt. Am I missing an obvious reason why it should removed?

Reply:
Missed the comment above. Removed.